src/java_bytecode/expr2java.h

```diff
@@ -50,4 +53,52 @@ class expr2javat:public expr2ct
 std::string expr2java(const exprt &expr, const namespacet &ns);
 std::string type2java(const typet &type, const namespacet &ns);
 
+/// Convert floating point number to a string without printing
+/// unnecessary zeros. Prints decimal if precision is not lost.
+/// Prints hexadecimal otherwise, and/or java-friendly NaN and Infinity
+template <typename float_type>
+std::string floating_point_to_java_string(float_type value)
+{
+  const auto is_float = std::is_same<float_type, float>::value;
+  static const std::string class_name = is_float ? "Float" : "Double";
+  if(std::isnan(value))
+    return class_name + ".NaN";
+  if(std::isinf(value) && value >= 0.)
+    return class_name + ".POSITIVE_INFINITY";
+  if(std::isinf(value) && value <= 0.)
+    return class_name + ".NEGATIVE_INFINITY";
+  // Using ostringstream to get string without trailing zeros
+  const std::string decimal = [&] { // NOLINT
+    std::ostringstream raw_stream;
+    raw_stream << value;
+    const auto raw_decimal = raw_stream.str();
+    if(raw_decimal.find('.') == std::string::npos)
+      return raw_decimal + ".0";
+    return raw_decimal;
+  }();
```

Comment:
Just curious: what is reason to use the lambda here? The following would seem more concise to me (just 5 lines instead of 8), but I could easily be wrong:
````c++
std::ostringstream raw_stream;
raw_stream << value;
if(raw_stream.str().find('.') == std::string::npos)
  raw_stream << ".0";
const std::string decimal = raw_stream.str();
````

Reply:
It is less code, but the lambda makes the mutable `raw_stream` inaccessible.

Reply:
I like to group values like that with lambdas. When reading code, we're usually asking ourselves the question of "where did this value come from". Grouping values in lambdas like this makes it obvious - the results of each "code block" are plainly visible and indented. Reduces the mental load, and should the lambda become too large - it can be easily detached into a discrete function.

Reply:
I guess that's where I tend to use blank lines. We probably agree to disagree on matters of style :-)